In [24]:
import numpy as np
from scipy import stats # importando scipy.stats
import pandas as pd # importando pandas
import matplotlib.pyplot as plt # importando matplotlib
import seaborn as sns
from scipy.stats import skew
import scipy.stats as st

# Pair Programming Estadística III

Vamos ahora a tomar de nuevo el conjunto de datos con el que 
estuvimos trabajando en los últimos días y vamos a continuar 
con el análisis de las estadísitcas de los contenidos del conjunto de datos.

Para todas los ejercicios de pair programming de
Estadística I, II y III usaremos el dataframe que 
se tome para el desarrollo de este ejercicio de pair programming.

Se ruega a la hora de realizar la entrega que incluyais el conjunto de datos que hayais decidido emplear para estos ejercicios.

# Objetivos
Los objetivos de hoy son:
 Extraer los siguentes estadísiticos del conjunto de datos para varias de las varibles disponibles:

### Tabla de contingencia

In [2]:
df_csv = pd.read_csv("01-imdb_top_1000.csv" )
df_csv.head(2)

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   

               Series_Title Released_Year Certificate  Runtime         Genre  \
0  The Shawshank Redemption          1994           A  142 min         Drama   
1             The Godfather          1972           A  175 min  Crime, Drama   

   IMDB_Rating                                           Overview  Meta_score  \
0          9.3  Two imprisoned men bond over a number of years...        80.0   
1          9.2  An organized crime dynasty's aging patriarch t...       100.0   

               Director          Star1           Star2       Star3  \
0        Frank Darabont    Tim Robbins  Morgan Freeman  Bob Gunton   
1  Francis Ford Coppola  Marlon Brando       Al Pacino  James Caan   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411

In [6]:
votos = df_csv['No_of_Votes']
meta = df_csv['Meta_score']
imdb = df_csv['IMDB_Rating']

In [7]:
votos

0      2343110
1      1620367
2      2303232
3      1129952
4       689845
        ...   
995     166544
996      34075
997      43374
998      26471
999      51853
Name: No_of_Votes, Length: 1000, dtype: int64

In [8]:
meta

0       80.0
1      100.0
2       84.0
3       90.0
4       96.0
       ...  
995     76.0
996     84.0
997     85.0
998     78.0
999     93.0
Name: Meta_score, Length: 1000, dtype: float64

In [3]:
df_csv["votos"] = pd.cut(df_csv["No_of_Votes"], bins= 100)
df_csv["meta_score"] = pd.cut(df_csv["Meta_score"], bins=100)

In [9]:

votos_crosstab = pd.crosstab(df_csv["meta_score"],
                            df_csv["votos"])
print(votos_crosstab)

votos            (22769.978, 48268.22]  (48268.22, 71448.44]  \
meta_score                                                     
(27.928, 28.72]                      0                     0   
(29.44, 30.16]                       0                     0   
(32.32, 33.04]                       0                     0   
(35.92, 36.64]                       0                     0   
(39.52, 40.24]                       1                     0   
...                                ...                   ...   
(95.68, 96.4]                        2                     2   
(96.4, 97.12]                        3                     3   
(97.84, 98.56]                       1                     0   
(98.56, 99.28]                       0                     0   
(99.28, 100.0]                       2                     1   

votos            (71448.44, 94628.66]  (94628.66, 117808.88]  \
meta_score                                                     
(27.928, 28.72]                     0  

### Coeficiente de correlación de las variables numéricas del dataset.

In [20]:
df_cor = np.corrcoef(df_csv["Meta_score"], df_csv["IMDB_Rating"])
df_cor

array([[nan, nan],
       [nan,  1.]])

### Sesgos de alguna de las variables numéricas del dataset.

In [23]:
sesgo_votos = skew(df_csv['No_of_Votes'])
print(sesgo_votos)

2.2965591310736615


### Intervalo de confianza de algunas de las variables numéricas del dataset

In [32]:
votos = df_csv['No_of_Votes']
imdb = df_csv['IMDB_Rating']

print(st.t.interval(alpha=0.99, df=len(votos)-1, loc=np.mean(votos), scale=st.sem(votos)))

print("El intervalo de confianza de votos al 99% es", np.mean(votos))

print(st.t.interval(alpha=0.99, df=len(imdb)-1, loc=np.mean(imdb), scale=st.sem(imdb)))

print("El intervalo de confianza de imdb al 99% es", np.mean(imdb))


(246975.76748985777, 300410.0545101423)
El intervalo de confianza de votos al 99% es 273692.911
(7.9268169461659586, 7.97178305383404)
El intervalo de confianza de imdb al 99% es 7.949299999999999


### Interpretación de los resultados.

Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo hincapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos.

Datos obtenidos:

- Tabla de contingencia

Hemos obtenido una tabla en la que cruzábamos la puntuación en meta con los votos obtenidos en cada película y se nos han cruzado dos intervalos. Para cada puntuación se han obtenido intervalos de aproximadamente 20000. Entre los rangos de 20000 y 1400000 es donde se han obtenido más votos.

- Coeficiente de correlación:

array([[nan, nan],
       [nan,  1.]])

Aquí la correlación no existe.

- Sesgo

2.2965591310736615

Al ser un sesgo positivo, nuetros datos estarían sesgados hacia la derecha, pero lo que realmente ocurre es que tenemos más películas de mayor metascore (las valoraciones son en decenas frente a IMDB que son unidades), no es que esté sesgada nuestra muestra, sino que es así.

- Intervalos de confianza:

    · Votos = 273692.911

       (246975.76748985777, 300410.0545101423)

    · IMDB = 7.949299999999

       (7.9268169461659586, 7.97178305383404)

Hemos elegido unos intervalos de confianza del 99% que quieren decir que según el número de votos lo más seguro es que estuviera entre 2046975 y 300410, ya que todos nuestros datos tienen números elevados.

Para el IMDB, lo más seguro es si le otorgan una calificación a una nueva película que introdujéramos, lo más seguro es que estaría entre esos valores.